src/block/Block.php

```diff
@@ -483,14 +483,19 @@ public function getEnchantmentTags() : array{
 	 * @param Item[] &$returnedItems Items to be added to the target's inventory (or dropped, if full)
 	 */
 	public function onBreak(Item $item, ?Player $player = null, array &$returnedItems = []) : bool{
-		$world = $this->position->getWorld();
-		if(($t = $world->getTile($this->position)) !== null){
-			$t->onBlockDestroyed();
-		}
-		$world->setBlock($this->position, VanillaBlocks::AIR());
+		$this->position->getWorld()->setBlock($this->position, VanillaBlocks::AIR());
```

Comment:
Noting this is probably BC breaking, since the behaviour is now changed for non-core consumers of `onBreak()`